### ISSUE TITLE
RB1 Xbox 360 drums can not be mapped properly on Linux.

### ISSUE THREAD

**Issue by DarkRTA** [EliteAsian123/YARG#237]
All of the inputs are showing up incorrectly and the game refuses to let me map my red pad.

**Comment by eckerj** [comment]
Make sure you have libhidapi-hidraw0 installed and that you have read/write permission to the /dev/hidraw* devices.

**Comment by TheNathannator** [comment]
I'm not sure that'll help for Xbox 360 controllers, those aren't handled through hidapi.

I need to get another custom backend in for Xbox 360 stuff on Linux (which would probably work for macOS as well). Without one there'll be no way for proper layouts to be provided there.

**Comment by DarkRTA** [comment]
> Make sure you have libhidapi-hidraw0 installed and that you have read/write permission to the /dev/hidraw* devices.

this is not the issue

**Comment by TadeLn** [comment]
The issue also affects Xbox 360 Xplorer 5-fret Guitars on Linux. Like on RB Drums, the button names are displayed incorrectly, and the red button cannot be detected. Additionally, the strum bar also cannot be mapped.

**Comment by qfoxb** [comment]
Wii LP with a raphnet also has improper button name detection, i just ignore it at this point

**Comment by TheNathannator** [comment]
Yeah, Unity's generic controller support does weird things with the button names, I have no clue why it screws them up lol

**Comment by coverslide** [comment]
I am experiencing this too. All other buttons and strum bar work, but red instrument pads for both RB1 guitar and drum kit and GH2 guitar from XBox360 have all failed to be detected in the bindings.

**Comment by Daanooo** [comment]
Any update on this?

**Comment by RileyTheFox** [comment]
No. We still need to write a new backend for XInput on Linux. Unfortunately it's quite a large task to do and there have been higher priority things to work on

**Comment by flavionm** [comment]
For anyone facing this issue, Steam Input is a viable workaround. It will detect the guitar (at least the wired RB1) as a regular controller, and then you can select the Steam Virtual Pad in game and map it.

**Comment by karl-run** [comment]
I think I'm also seeing this issue on YARG 0.14 on Bazzite. During startup I see that YARG detects the Xbox 360 Xplorer in the top right corner, but I'm not able to select it as a device on a profile. I've configured the udev rules as per the README.

For what it's worth, the Rock Band The Beatles wireless guitar through the Xbox 360 Wireless Dongle configured a profile automatically out of the box. 😄 

Edit: I also see the Xplorer in the "game controller" settings in Bazzite itself, and all the buttons and axis' works.

Edit 2: This YouTrack issue seems very relevant: https://yarg.youtrack.cloud/issue/YARG-167/Xbox-360-controllers-on-Linux-do-not-work-correctly

**Comment by shibeobi12** [comment]
Just throwing myself in the ring as well, GH WT guitar (with slide touchpad) and a Raphnet adapter also cannot map the red button input no matter what I've been trying. This is on CachyOS, KDE.

I tried Steam's virtual input, but it doesn't allow me to properly map the up strum or whammy. :/ 

Edit 8/31: I updated the firmware of the Raphnet and on 2.2.10 it exposes a secondary device of some kind that allowed me to map. Seems to be called 1 player WUSBMEM2.2 and appeared post firmware update. That one seems to work!
